### ISSUE TITLE
autocmd raises E685 in debug mode

### ISSUE THREAD

**Issue by lacygoill** [neovim/neovim#11273]
- `nvim --version`:        NVIM v0.5.0-112-g2e4465e05 Build type: Release
 - Vim (version:  8.1 Included patches: 1-2197) behaves differently? Yes
 - Operating system:     Ubuntu 16.04.6 LTS
 - Terminal emulator:    XTerm(322)
 - `$TERM`:                xterm-256color
 - Shell:                zsh 5.7.1 (x86_64-pc-linux-gnu)

### Steps to reproduce

Run this shell command:

    echo n | nvim -Nu NONE -es +'au CmdlineEnter > let a=1' +'set vbs=1 | debug b'

### Actual behaviour

`E685` is raised.

    E5500: autocmd has thrown an exception: Vim(let):E685: Internal error: hash_add()

### Expected behaviour

No error is raised.

### Additional information

According to `:h E685`:

    This is an internal error.  If you can reproduce it, please send in a bug report.

this is an internal error which should be reported.

---

I'm not sure whether it has been forgotten, or it's not meant to be documented, but I can't find anything about `E5500` in the help (`:helpg E5500` finds no match).

---

This doesn't happen in Vim:

    $ vim -Nu NONE +'au CmdlineEnter > let a=1' +'debug b'
    >next

**Comment by ffanzhang** [comment]
This is the only place that produces hash_add() error.
in src/nvim/hashtab.c
line 206
```
int hash_add(hashtab_T *ht, char_u *key)
{
  hash_T hash = hash_hash(key);
  hashitem_T *hi = hash_lookup(ht, (const char *)key, STRLEN(key), hash);
  if (!HASHITEM_EMPTY(hi)) {
    internal_error("hash_add()");
    return FAIL;
  }
  hash_add_item(ht, hi, key, hash);
  return OK;
}
```
The caller of hash_add() could exist in src/nvim/eval.c, src/nvim/eval/typval.c, or src/nvim/spellfile.c

**Comment by zeertzjq** [comment]
Seems fixed between v0.6.0 and v0.7.0
